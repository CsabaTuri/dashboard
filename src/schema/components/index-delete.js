import React from "react"
import { connect } from "react-redux"
import { browserHistory } from "react-router"

import DeleteForm from "./delete-form"
import { selectedIndex, selectedDatabase, deleteIndex } from "../"
import { buildResourceUrl } from "../../router"
import { faunaClient } from "../../authentication"
import { notify } from "../../notifications"

export const IndexDelete = ({ client, index, path, databaseUrl }) => {
  const onDelete = () => notify("Index deleted successfully", dispatch =>
    dispatch(deleteIndex(client, path, index.get("name"))).then(() =>
      browserHistory.push(buildResourceUrl(databaseUrl, "indexes"))
    )
  )

  return <DeleteForm
    buttonText="Delete Index"
    type="index"
    title={`Delete ${index.get("name")}`}
    validateName={index.get("name")}
    onDelete={onDelete} />
}

export default connect(
  state => ({
    client: faunaClient(state),
    index: selectedIndex(state),
    path: selectedDatabase(state).get("path"),
    databaseUrl: selectedDatabase(state).get("url")
  })
)(IndexDelete)
