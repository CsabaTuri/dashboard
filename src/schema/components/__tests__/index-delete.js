import React from "react"
import { Map } from "immutable"
import { shallow } from "enzyme"
import { shallowToJson } from "enzyme-to-json"

import { IndexDelete } from "../index-delete"

jest.mock("../../../notifications", () => ({
  notify: (msg, fn) => fn(() => Promise.resolve())
}))

jest.mock("react-router", () => ({
  browserHistory: { push: jest.fn() }
}))

const { browserHistory } = require("react-router")

describe("IndexDelete Component", () => {
  let comp

  beforeEach(() => {
    const index = Map.of("name", "a-index")
    comp = shallow(<IndexDelete index={index} databaseUrl="/db/databases" />)
  })

  it("display delete form", () => {
    expect(shallowToJson(comp)).toMatchSnapshot()
  })

  it("deletes the index", () => {
    return comp.find("Connect(DeleteForm)").props().onDelete().then(() => {
      expect(browserHistory.push).toHaveBeenCalledWith("/db/indexes")
    })
  })
})
