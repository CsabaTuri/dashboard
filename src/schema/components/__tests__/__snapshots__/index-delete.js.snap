// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IndexDelete Component display delete form 1`] = `
<Connect(DeleteForm)
  buttonText="Delete Index"
  onDelete={[Function]}
  title="Delete a-index"
  type="index"
  validateName="a-index"
/>
`;
